gcc/testsuite/rust/compile/issue-852.rs

```diff
@@ -0,0 +1,30 @@
+// { dg-options "-w" }
+extern "C" {
+    fn printf(s: *const i8, ...);
+}
+
+enum Foo {
+    A,
+    B(i32),
+}
+
+fn main() {
+    let result = Foo::B(123);
+
+    match result {
+        A => unsafe {
```

Comment:
```suggestion
        Foo::A => unsafe {
```

Reply:
Otherwise the second match arm should never be entered - I didn't catch up on the Zulip discussion which addresses this tho :D